src/core/ddsc/src/dds__data_allocator.h

```diff
@@ -37,6 +38,7 @@ typedef struct dds_iox_allocator {
     iox_pub_t pub;
     iox_sub_t sub;
   } ref;
+  ddsrt_mutex_t mutex;
```

Comment:
I ran into one issue trying to build it locally on macOS with Iceoryx: the mutex makes it too large on macOS making the static assertion fail. The patch below takes care of that:
```diff
diff --git a/src/core/ddsc/include/dds/ddsc/dds_data_allocator.h b/src/core/ddsc/include/dds/ddsc/dds_data_allocator.h
index b8bfc535..123c3b8a 100644
--- a/src/core/ddsc/include/dds/ddsc/dds_data_allocator.h
+++ b/src/core/ddsc/include/dds/ddsc/dds_data_allocator.h
@@ -23,7 +23,9 @@
 extern "C" {
 #endif
 
-#define DDS_DATA_ALLOCATOR_MAX_SIZE (8 * sizeof (void *))
+// macOS' mutexes require quite a lot of space, but it is not quite enough
+// to make this system-dependent
+#define DDS_DATA_ALLOCATOR_MAX_SIZE (12 * sizeof (void *))
 
 typedef struct dds_data_allocator {
   dds_entity_t entity;
```

Reply:
Done. It is kind of error-prone to do this if we want to implement this "base class" allocator interface. Should maybe have use the "first member of a derived is the base struct" pseudo-inheritance if we intend to keep the allocator anyway.